firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteStore.java

```diff
@@ -278,13 +307,17 @@ public void handleCredentialChange() {
       // for the new user and re-fill the write pipeline with new mutations from the LocalStore
       // (since mutations are per-user).
       Logger.debug(LOG_TAG, "Restarting streams for new credential.");
-      networkEnabled = false;
-      disableNetworkInternal();
-      onlineStateTracker.updateState(OnlineState.UNKNOWN);
-      enableNetwork();
+      restartNetwork();
     }
   }
 
+  private void restartNetwork() {
```

Comment:
Consider moving this closer to the other network-controlling methods like `disableNetworkInternal` so that it's more obvious that a choice is available.

Reply:
Done.